client/src/telethon/_impl/mtsender/sender.py

```diff
@@ -323,11 +296,7 @@ def _on_net_read(self, read_buffer: bytes) -> None:
             else:
                 del self._read_buffer[:n]
                 self._process_mtp_buffer()
-
-    def _on_net_write(self) -> None:
-        for req in self._requests:
-            if isinstance(req.state, Serialized):
-                req.state = Sent(req.state.msg_id, req.state.container_msg_id)
+                self._response_event.set()
```

Comment:
Why do we set this here? This event is only intended to wake up the task when there's a new request ready to be sent.

Reply:
The step should be completed when the response is received